includes/models/model.llms.access.plan.php

```diff
@@ -135,6 +143,72 @@ protected function get_creation_args( $title = '' ) {
 
 	}
 
+	/**
+	 * Retrieve the full URL to redirect to after successful checkout
+	 * @return   string
+	 * @since    [version]
+	 * @version  [version]
+	 */
+	public function get_redirection_url() {
+
+		// what type of redirection is set up by user?
+		$redirect_type = $this->get( 'checkout_redirect_type' );
+
+		// by default, no special redirection is needed.
+		$redirection = '';
+
+		$available = $this->is_available_to_user( get_current_user_id() );
+
+		$query_redirection = filter_input( INPUT_GET, 'redirect', FILTER_VALIDATE_URL );
+
+		// force redirect querystring parameter over all else.
+		if ( ! empty( $query_redirection ) ) {
+			$redirection = $query_redirection;
+
+		} else {
+
+			if ( ! $available && 'no' === $this->get( 'checkout_redirect_forced' ) ) {
+				$redirect_type = 'membership';
+			}
+
+			switch ( $redirect_type ) {
+
+				// redirect to itself
+				case 'self':
+					/* Only set up when it is a member's only access plan with forced redirection to course.
+					* This will ensure that on a regular access plan, no special parameter is added to querystring.
+					* At the same time, if it is a members' only access plan,
+					* after membership checkout we'd like to force redirect to course
+					*/
+					if ( ! $available && 'yes' === $this->get( 'checkout_redirect_forced' ) ) {
+						$redirection = get_permalink( $this->get( 'product_id' ) );
+					}
+					break;
+				case 'page':
+					$redirection = get_permalink( $this->get( 'checkout_redirect_page' ) );
+					break;
+				case 'url':
+					$redirection = $this->get( 'checkout_redirect_url' );
+					break;
+				case 'membership':
+				default:
+					break;
+			}
+		}
+
+		/**
+		 * Filter the checkout redirection parameter
+		 *
+		 * @param    string               $redirection The calculated url to redirect to.
+		 * @param    string               $redirection_type Available redirection types 'self', 'membership', 'page', 'url' or a custom type.
+		 * @param    LLMS_Acccess_Plan    $this Current Access Plan object.
+		 * @since    [version]
+		 * @version  [version]
+		 */
+		return urlencode( apply_filters( 'llms_plan_get_checkout_redirection', $redirection, $redirection_type, $this ) );
```

Comment:
Unit tests seem to be failing because this variable should actually be `$redirect_type` as defined on line 155 above